topdown/resolver.go

```diff
@@ -66,6 +66,7 @@ func (t *resolverTrie) mktree(ctx context.Context, in resolver.Input) (ast.Value
 	}
 	obj := ast.NewObject()
 	for k, child := range t.children {
+		in = resolver.Input{Ref: append(in.Ref.Copy(), ast.NewTerm(k)), Input: in.Input}
```

Comment:
@tsandall curious if you have any suggestions on this, I'm not 100% sure the original intent of `Input.Ref`, if it was supposed to be the original one the resolver trie is trying to resolve or the ref that the individual resolver was being asked to resolve. Doing this copy/append business doesn't seem ideal, we could front load it when building the trie maybe?